From e152d67004977b02eec7fb8230d044913b9b0871 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:24:23 -0400
Subject: [cpufreq] intel_pstate: Make intel_pstate_kobject and debugfs_parent locals

Message-id: <1409235879-1283-170-git-send-email-prarit@redhat.com>
Patchwork-id: 89494
O-Subject: [RHEL7.1 PATCH BZ 1134639 169/185] cpufreq: intel_pstate: Make intel_pstate_kobject and debugfs_parent locals
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 317dd50e80775434a2ea593ad8522e728ed94e9d
Author: Stratos Karafotis <stratosk@semaphore.gr>
Date:   Fri Jul 18 08:37:17 2014 -0700

    cpufreq: intel_pstate: Make intel_pstate_kobject and debugfs_parent locals

    Since we never remove sysfs entry and debugfs files, we can make
    the intel_pstate_kobject and debugfs_parent locals.

    Also, annotate with __init intel_pstate_sysfs_expose_params()
    and intel_pstate_debug_expose_params() in order to be freed
    after bootstrap.

    Signed-off-by: Stratos Karafotis <stratosk@semaphore.gr>
    Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 86631cb..601c428 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -253,9 +253,9 @@ static struct pid_param pid_files[] = {
  {NULL, NULL}
 };
 
-static struct dentry *debugfs_parent;
-static void intel_pstate_debug_expose_params(void)
+static void __init intel_pstate_debug_expose_params(void)
 {
+ struct dentry *debugfs_parent;
  int i = 0;
 
  debugfs_parent = debugfs_create_dir("pstate_snb", NULL);
@@ -342,10 +342,10 @@ static struct attribute *intel_pstate_attributes[] = {
 static struct attribute_group intel_pstate_attr_group = {
  .attrs = intel_pstate_attributes,
 };
-static struct kobject *intel_pstate_kobject;
 
-static void intel_pstate_sysfs_expose_params(void)
+static void __init intel_pstate_sysfs_expose_params(void)
 {
+ struct kobject *intel_pstate_kobject;
  int rc;
 
  intel_pstate_kobject = kobject_create_and_add("intel_pstate",
-- 
1.7.1